### PR TITLE
feat(table): idempotent list and typing

### PR DESCRIPTION
**Pull request by bdebon**
# What does this PR do?

[Link to the JIRA ticket](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-631)

- ordering alphabetically breadcrumb and table
- table stop ordering alphabetically if we manually click on a sorting (eg updated_at)
- remove a lot of any from the table component in profit of Generic typing
- Added a util to sort by one of the key of the Generic
- Used this util in every getByXXId methods from slices. We can choose the key we want the sort to happen
- Fixed the glitch with filtering in MenuGroup for all the menugroup of the console
- Ordering organization list

CleanShot 2023-02-21 at 17 29 58@2x

CleanShot 2023-02-22 at 12 07 58@2x

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(table): idempotent list and typing - 2023-02-21T16:31:10Z](https://new.console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/a1ed506f-0017-4918-8efd-38767bc22c86/services/general)

 Another comment will be posted when deployments are terminated

**Comment by evoxmusic** [comment]
Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [[PR] staging - feat(table): idempotent list and typing - 2023-02-21T16:31:10Z](https://new.console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/a1ed506f-0017-4918-8efd-38767bc22c86/services/logs)

**Review by RemiBonnet — COMMENTED** [review]
Well done @bdebon, my feedback on comments

File: libs/shared/console-shared/src/lib/buttons-actions/ui/application-buttons-actions/application-buttons-actions.tsx [hunk 1]
```diff
@@ -246,7 +246,10 @@ export function ApplicationButtonsActions(props: ApplicationButtonsActionsProps)
       }
     }
 
-    setButtonStatusActions([{ items: topItems }, { items: bottomItems }])
+    setButtonStatusActions([
+      { items: topItems, dontSortAlphabetically: true },
+      { items: bottomItems, dontSortAlphabetically: true },
```
[the remark this answers is not in the document]
Comment (bdebon) [reply libs/shared/console-shared/src/lib/buttons-actions/ui/application-buttons-actions/application-buttons-actions.tsx:251]: Okay, gonna try to reverse the behavior yes, not sorted by default.

File: libs/shared/ui/src/lib/components/table/table-head-filter/table-head-filter.tsx [hunk 2]
```diff
@@ -7,20 +7,20 @@ import Menu from '../../menu/menu'
...
 
-export interface TableHeadFilterProps {
+export interface TableHeadFilterProps<T> {
   title: string
-  dataHead: TableHeadProps
+  dataHead: TableHeadProps<T>
   currentFilter: string
   setCurrentFilter: Dispatch<SetStateAction<string>>
-  defaultData: any[]
+  defaultData: T[]
```
[the remark this answers is not in the document]
Comment (bdebon) [reply libs/shared/ui/src/lib/components/table/table-head-filter/table-head-filter.tsx:15]: Yes! I was happy with the final result! And the autocompletion are much better now when we use table component!

File: libs/pages/services/src/lib/ui/page-general/page-general.tsx [hunk 3]
```diff
@@ -63,6 +63,7 @@ function PageGeneralMemo(props: PageGeneralProps) {
           setDataSort={setData}
           className="mt-2 bg-white rounded-sm flex-grow overflow-y-auto min-h-0"
           columnsWidth="30% 20% 25% 20%"
+          defaultSortingKey={'name'}
```
[the remark this answers is not in the document]
Comment (bdebon) [reply libs/pages/services/src/lib/ui/page-general/page-general.tsx:66]: yup sorry!

[the remark this answers is not in the document]
Comment (bdebon) [reply libs/pages/services/src/lib/ui/page-general/page-general.tsx:66]: But you'll see, so conveniant thanks to the generic type, defaultSortingKey is autosuggested with the keys of your entity. Super nice for the DE!

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**